### PR TITLE
blog: announcing plug-and-play extensions

### PR DESCRIPTION
**Pull request by nicks**
Hello @ellenkorbes,

Please review the following commits I made in branch nicks/extensions:

9c793357a98b9ed02eaee91b27d199999009823b (2021-08-24 20:50:26 -0400)
blog: announcing plug-and-play extensions

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
I want to cannibalize this into a new extensions doc page, but first wanted to do an announcement blog post that got more at the underlying philosophy and why we're building it this way (since, at first blush, the "new" extension system is more limited than the old one)

File: blog/_posts/2021-08-25-plug-and-play-extensions.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+then wrote a server that translated `PinterestService` into low-level objects.
+
+I've been thinking a lot about servers as a tool, and how we can use it at Tilt.
+
+## Extending Tilt With More Config
+
+Lots of teams want to extend Tilt to support more types of dev environments. We
+gave them the ability to add new functions to their Tiltfile.
+
+Here's a good example of a Tilt extension: [`git_resource`](https://github.com/tilt-dev/tilt-extensions/tree/master/kubefwd). From the README:
```
Comment (ellenkorbes) [inline blog/_posts/2021-08-25-plug-and-play-extensions.md:55]:
```suggestion
Here's a good example of a Tilt extension: [`git_resource`](https://github.com/tilt-dev/tilt-extensions/tree/master/git_resource). From the README:
```

**Comment by ellenkorbes** [comment]
This post was very hard to follow. The first 50% of it is backstory, and it never tells me where the story is going, or why I should care and what to expect. When it finally gets to what you were trying to get to, it's over before it answers the questions that it raised. What in practice does this look like? What is it bad for? What is the mental checklist to tell when I should use this technique versus any of the others? I feel there's too much history and not enough present.

**Comment by nicks** [comment]
ya, that's fair. I think the general problem is that we don't have many extensions in this form yet. So ended up talking more about how it's implemented and why we implemented it that way than about how to use it right now.

i didn't put a lot of work into this, and would be fine trashing it and/or starting over.

**Comment by nicks** [comment]
it might work better if we just trash the first section entirely.

**Comment by nicks** [comment]
ahhh i fucked up the merge
